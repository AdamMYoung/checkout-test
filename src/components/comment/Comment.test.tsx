import { render, screen } from '@testing-library/react';
import { axe } from 'jest-axe';
import { Comment, CommentAuthor, CommentDate, CommentRating, CommentText } from '..';

describe('CommentAuthor', () => {
    it('passes aXe validation', async () => {
        const { container } = render(<CommentAuthor>Test</CommentAuthor>);

        expect(await axe(container)).toHaveNoViolations();
    });

    it('matches snapshot', () => {
        const { container } = render(<CommentAuthor>Test</CommentAuthor>);

        expect(container).toMatchSnapshot();
    });

    it('renders all children matching the correct role', () => {
        const children = 'Hello World!';
        render(<CommentAuthor>{children}</CommentAuthor>);

        const author = screen.getByRole('heading', { name: children });

        expect(author).toBeInTheDocument();
    });
});

describe('CommentDate', () => {
    it('passes aXe validation', async () => {
        const { container } = render(<CommentDate>Test</CommentDate>);

        expect(await axe(container)).toHaveNoViolations();
    });

    it('matches snapshot', () => {
        const { container } = render(<CommentDate>Test</CommentDate>);

        expect(container).toMatchSnapshot();
    });

    it('renders all children matching the correct role', () => {
        const children = 'Hello World!';
        render(<CommentDate>{children}</CommentDate>);

        const date = screen.getByText(children);

        expect(date).toBeInTheDocument();
    });
});

describe('CommentRating', () => {
    it('passes aXe validation', async () => {
        const { container } = render(<CommentRating value={4} />);

        expect(await axe(container)).toHaveNoViolations();
    });

    it('matches snapshot', () => {
        const { container } = render(<CommentRating value={4} />);

        expect(container).toMatchSnapshot();
    });

    it('renders the correct rating provided', () => {
        const rating = 4;
        render(<CommentRating value={rating} />);

        const ratingElement = screen.getByLabelText('4 out of 5 stars');

        expect(ratingElement).toBeInTheDocument();
    });
});

describe('CommentText', () => {
    it('passes aXe validation', async () => {
        const { container } = render(<CommentText>Test</CommentText>);

        expect(await axe(container)).toHaveNoViolations();
    });

    it('matches snapshot', () => {
        const { container } = render(<CommentText>Test</CommentText>);

        expect(container).toMatchSnapshot();
    });

    it('renders all children matching the correct role', () => {
        const children = 'Hello World!';
        render(<CommentText>{children}</CommentText>);

        const text = screen.getByText(children);

        expect(text).toBeInTheDocument();
    });
});

describe('Comment', () => {
    it('passes aXe validation', async () => {
        const { container } = render(<Comment>Test</Comment>);

        expect(await axe(container)).toHaveNoViolations();
    });

    it('matches snapshot', () => {
        const { container } = render(<Comment>Test</Comment>);

        expect(container).toMatchSnapshot();
    });

    it('renders as the correct role', () => {
        render(<Comment />);

        const comment = screen.getByRole('article');

        expect(comment).toBeInTheDocument();
    });
});
