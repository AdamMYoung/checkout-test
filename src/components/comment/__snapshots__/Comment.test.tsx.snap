// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment matches snapshot 1`] = `
<div>
  <article
    class="chakra-stack css-n21gh5"
  />
</div>
`;

exports[`CommentAuthor matches snapshot 1`] = `
<div>
  <h2
    class="chakra-heading css-1tapsqs"
  >
    Test
  </h2>
</div>
`;

exports[`CommentDate matches snapshot 1`] = `
<div>
  <p
    class="chakra-text css-5ft5pb"
  >
    Test
  </p>
</div>
`;

exports[`CommentRating matches snapshot 1`] = `
<div>
  <span
    aria-label="4 out of 5 stars"
    class="chakra-stack css-193rmy8"
  >
    <svg
      class="chakra-icon css-1k73fkr"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
        fill="currentColor"
      />
    </svg>
    <svg
      class="chakra-icon css-1k73fkr"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
        fill="currentColor"
      />
    </svg>
    <svg
      class="chakra-icon css-1k73fkr"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
        fill="currentColor"
      />
    </svg>
    <svg
      class="chakra-icon css-1k73fkr"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
        fill="currentColor"
      />
    </svg>
    <svg
      class="chakra-icon css-6aos0a"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
        fill="currentColor"
      />
    </svg>
  </span>
</div>
`;

exports[`CommentText matches snapshot 1`] = `
<div>
  <p
    class="chakra-text css-0"
  >
    Test
  </p>
</div>
`;
